dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaChatCompletionTests.cs

```diff
@@ -0,0 +1,183 @@
+        //Arrange
+        var sut = new OllamaChatCompletionService(
+            "fake-model",
+            new Uri("http://localhost:11434"),
+            httpClient: this._httpClient);
+
+        var chat = new ChatHistory();
+        chat.AddMessage(AuthorRole.User, "fake-text");
+
+        //Act
+        var messages = await sut.GetChatMessageContentsAsync(chat);
+
+        //Assert
+        Assert.NotNull(messages);
+
+        var message = messages.SingleOrDefault();
+        Assert.NotNull(message);
+        Assert.Equal("This is test completion response", message.Content);
+    }
+
+    [Fact]
+    public async Task GetChatMessageContentsShouldHaveModelAndMetadataAsync()
+    {
+        //Arrange
+        var sut = new OllamaChatCompletionService(
+            "phi3",
+            new Uri("http://localhost:11434"),
+            httpClient: this._httpClient);
+
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+        {
+            Content = new StringContent(File.ReadAllText("TestData/chat_completion_test_response.txt"))
+        };
+
+        var chat = new ChatHistory();
+        chat.AddMessage(AuthorRole.User, "fake-text");
+
+        //Act
+        var messages = await sut.GetChatMessageContentsAsync(chat);
+
+        //Assert
+        Assert.NotNull(messages);
+        var message = messages.SingleOrDefault();
+        Assert.NotNull(message);
+
+        // Assert
+        Assert.NotNull(message.ModelId);
+        Assert.Equal("phi3", message.ModelId);
+    }
+
+    [Fact]
+    public async Task GetStreamingChatMessageContentsShouldHaveModelAndMetadataAsync()
+    {
+        //Arrange
+        var expectedModel = "phi3";
+        var sut = new OllamaChatCompletionService(
+            expectedModel,
+            new Uri("http://localhost:11434"),
+            httpClient: this._httpClient);
+
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+        {
+            Content = new StreamContent(File.OpenRead("TestData/chat_completion_test_response_stream.txt"))
+        };
+
+        var chat = new ChatHistory();
+        chat.AddMessage(AuthorRole.User, "fake-text");
+
+        // Act
+        StreamingChatMessageContent? lastMessage = null;
+        await foreach (var message in sut.GetStreamingChatMessageContentsAsync(chat))
+        {
+            lastMessage = message;
+            Assert.NotNull(message.Metadata);
+        }
+
+        // Assert
+        Assert.NotNull(lastMessage!.ModelId);
+        Assert.Equal(expectedModel, lastMessage.ModelId);
+
+        Assert.IsType<OllamaMetadata>(lastMessage.Metadata);
+        var metadata = lastMessage.Metadata as OllamaMetadata;
+        Assert.NotNull(metadata);
+        Assert.NotEmpty(metadata);
+        Assert.True(metadata.Done);
+    }
+
+    public void Dispose()
+    {
+        this._httpClient.Dispose();
+        this._messageHandlerStub.Dispose();
+    }
+}
```

Comment:
Can you add some tests to validate the request content, in particular that prompt execution settings are included correctly.